version: "3.8"

services:
  discovery:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: discovery-api
    restart: unless-stopped
    ports:
      - "15002:15002"
    # networks:
    #   - jupyterhub-network
    environment:
      REDIS_HOST: redis
      REDIS_PORT: 16379
      REDIS_PASSWORD: "redis@pass"
    depends_on:
      - redis
  redis:
    image: redis:7-alpine
    container_name: redis
    restart: unless-stopped
    volumes:
      - data:/data
      - ./redis.conf:/usr/local/etc/redis/redis.conf
    command: ["redis-server", "/usr/local/etc/redis/redis.conf"]
    ports:
      - "16379:16379"
    # networks:
    #   - jupyterhub-network
    healthcheck:
      test: ["CMD", "redis-cli", "-p", "16379", "ping"]
      interval: 10s
      timeout: 3s
      retries: 3

volumes:
  data:

# networks:
#   jupyterhub-network:
#     name: jupyterhub-network


